File: get-mock.js
const fs = require('fs');

console.time('Generate time');
const consoleOutput = false;

const count = process.argv[2];
const locale = process.argv[3];
const typoCount = process.argv[4];

const result = [];
let delState = 1; // To keep balace. +1: delete char, -1: paste char

const {
  address,
  name,
  additional
} = JSON.parse(fs.readFileSync(`./lib/locales/${locale}.json`, 'utf8'));

for (let i = 0; i < count; i++) {
  const mock = [getName(name), getAddress(address), getPhoneNumber(additional.phonenumber)];
  if (typoCount != 0) {
    for (let i = 0; i < typoCount; i++) {
      const mockPart = getRandom(mock.length);
      mock[mockPart] = getTypo(mock[mockPart]);
    }
  }
  result.push(mock.join('| '));
}
consoleOutput ? result.forEach(el => console.log(el)) : fs.appendFileSync('./output.csv', result.join('\n'), err => console.error(err));

function getTypo(str) {
  const missingIndex = getRandom(str.length - 3, 1);
  switch (getRandom(2)) {
    case 0: // delete or paste char
      delState *= -1;
      return `${str.slice(0, missingIndex)}${str.slice(missingIndex + delState)}`;
    case 1: // change order
      return `${str.slice(0, missingIndex)}${str[missingIndex+1]}${str[missingIndex]}${str.slice(missingIndex+2)}`;
    default:
      break;
  }
}

function getName(obj) {
  const keys = Object.keys(obj);
  const gender = keys[getRandom(keys.length)];
  const randomName = [];
  for (const field in obj[gender]) {
    const target = obj[gender][field];
    const name = target[getRandom(target.length)]
    randomName.push(name);
  }
  return randomName.join(' ');
}

function getAddress(obj) {
  const randomAddress = [];
  for (const field in obj) {
    const target = obj[field];
    const name = target[getRandom(target.length)];
    randomAddress.push(`${name}${field === 'prefix' ? ' ' : ', '}`);
  }
  randomAddress.push(getRandom(200), getPostfix());
  return randomAddress.join('');
}

function getPostfix() {
  const chars = ['-', '/', ' '];
  if (getRandom(2)) {
    return `${chars[getRandom(2)]}${getRandom(2) ? additional.postfix[getRandom(5)] : getRandom(5, 1)}`;
  }
}

function getPhoneNumber(pattern) {
  return pattern.split('').map((el, i, arr) => el === '#' ? arr[i] = getRandom(9) : el).join('');
}

function getRandom(max, min = 0) {
  return Math.floor(Math.random() * max) + min;
}

console.timeEnd('Generate time');

File: get-mock.rb
require "json"

if (Gem.win_platform?)
  Encoding.default_external = Encoding.find(Encoding.locale_charmap)
  Encoding.default_internal = __ENCODING__
  [STDIN, STDOUT].each do |io|
    io.set_encoding(Encoding.default_external, Encoding.default_internal)
  end
end

@time_start = Time.now.getutc

@length = ARGV[0].to_i
@locale = ARGV[1]
@typo_per_line = ARGV[2].to_i
@typo_per_result = ((ARGV[2].to_f % 1)*@length).round

@file = File.open("./lib/locales/#{@locale}.json", "r:UTF-8", &:read)
@data = JSON.parse @file
@address = @data['address']
@name = @data['name']
@additional = @data['additional']
@phone_pattern = @additional['phonenumber']

@result = []

@length.times do 
  @address_arr = []
  @name_arr = []
  @phone_arr = Marshal.load( Marshal.dump( @phone_pattern.sample.split('') ) )

  @address.each { | key, value |
  if (value[0] == '#') then
    if (rand(2) == 1 && key == 'build_num') then 
      @chars = ['-', '/', ' '];
      @address_arr.push(rand(250), @chars.sample, @additional['postfix'].sample)
    else
      @address_arr.push(rand(250))
    end
  else
    @address_arr.push(value.sample)
  end

  if (key.include? 'prefix') then
    @address_arr.push(' ')
  else
    @address_arr.push(', ')
  end
  }

  @name[@name.keys.sample].each_value { | value |
      @name_arr.push(value.sample)
  }

  @phone_arr.each_with_index { | value, index |
  if (value === '#') then
    @phone_arr[index] = rand(9)
  end
  }

  @pre_result = [@name_arr.join(' '), @address_arr.join, @phone_arr.join('')]
  @del_state = 1 # To keep balace. +1: delete char, -1: paste char

  def getMistake (str)
    @mistake = rand(str.length - 3) + 1;
    case (rand(1))
      when 0 # delete or paste char
        @del_state *= -1;
        return "#{str[0...@mistake]}#{str[(@mistake + @del_state)...str.length]}";
      when 1 # change order
        return "#{str[0...@mistake]}#{str[@mistake+1]}#{str[@mistake]}#{str[(@mistake+2)...str.length]}";
      end
  end

  if (@typo_per_line > 0) then
    @typo_per_line.times do 
      @typo_index = rand(@pre_result.length)
      @pre_result[@typo_index] = getMistake(@pre_result[@typo_index])
    end
  end

  if (@typo_per_result > 0 && (rand() > 0.7 || (@length - @result.length) < @typo_per_result) ) then
    @typo_index = rand(@pre_result.length)
      @pre_result[@typo_index] = getMistake(@pre_result[@typo_index])
    @typo_per_result -=1
  end
  @result.push(@pre_result.join(' | '))
end

# @result.each { |x| puts x }

p "Generate in #{Time.now.getutc - @time_start}ms"